[feature] Document View logs

shridarpatil opened the pull request:
Log user when ever he views a document


![image](https://user-images.githubusercontent.com/11792643/38977826-25896400-43d3-11e8-90c4-fcc57293a6f6.png)



![image](https://user-images.githubusercontent.com/11792643/38977749-eb19146e-43d2-11e8-8b4e-c843d7f842f9.png)

lasalesi committed:
using regex instead of soup to parse pdf page margins

lasalesi committed:
corrected trailing whitespace (codacy)

tundebabzy committed:
move Subscription to frappe as Auto Repeat

tundebabzy committed:
fix test case

tundebabzy committed:
appease codacy

tomtom5152 committed:
Migrate password hashing away from mysql password()

This is deprecated and needs to be replaced. 
Use passlib to hash, store, verify and upgrade as necessary.
Includes patch to migrate existing passwords in a non-breaking way.

Fixes #5195

lasalesi committed:
added test case

lasalesi committed:
removed trailing whitespaces (codacy)

tomtom5152 committed:
Fix code quality issue

Trailing whitespace in patch SQL, was unoticable in editor.

tomtom5152 committed:
Fix PR issues from #5200.

Fix issue with with rebase messing up patches.txt.

Change auth parameter back to user.

rmehta committed:
Update migrate_passwords_passlib.py

"code is poetry"

rmehta committed:
Update password.py

shridarpatil committed:
Added track view checkbox in doctype

shridarpatil committed:
Add comment to communication whenever user views a document

shridarpatil committed:
typo fixed

shridarpatil committed:
add view log after permission check

netchampfaris committed:
Disable "User Settings" in query reports (#5473)

rmehta commented on frappe/model/document.py:
```diff
@@ -1143,6 +1143,23 @@ def add_seen(self, user=None):
+			user = frappe.session.user
+
+		if self.meta.track_views:
+			frappe.get_doc({
```
not sure if communication is the right table for this... shall we create another table to partition the info?

shridarpatil replied:
The structure of the new doctype should be same as communication doctype?

rmehta replied:
No can be much smaller! How about `View Log` ?

Zlash65 committed:
fix user-tags trigger (#5471)

shreyashah115 committed:
Rename doctype Standard Reply to Email Template (#5444)

* Rename Standard Reply to Email Template

* rename occurrences

Commit:
Fix autoname in data_import.py (#5474)

ryuliantoro committed:
Update permission_manager_help.html (#5462)

translation tag

netchampfaris committed:
Show Permission Restrictions in a Table (#5457)

KanchanChauhan committed:
[Fix] Print for Report View (#5470)

* [Fix] Print for List report view fixed

* Update report_view.js

* Show standard filters too

Zlash65 committed:
[Fix] Fix treeview adding child to wrong parent and Freeze screen while adding a node. (#5464)

* fix parent wrong selected, freeze screen while adding a node

* unfreeze dom always

* fix translation method

surajshetty3416 committed:
Nested permission (#5460)

* nested permission

* test for nested permission

* [minor] Format code

* Fix nested permission test

* Fix codacy

Zlash65 committed:
minor user tags fix

Commit:
fixed the letterhead (#5477)

Zlash65 committed:
multiselect - behave like link field

netchampfaris committed:
Listview fixes (#5479)

* update frappe-datatable to 0.0.5

* [fix] List view realtime update

nabinhait committed:
Minor fix in reloading email template

ashish-greycube committed:
Update trigger-event-on-deletion-of-grid-row.md (#5488)

Trigger name is incorrect, it should be "before_items_remove"

saurabh6790 committed:
bumped to version 10.1.25

pratu16x7 committed:
[fix] query report dict key fetch

codingCoffee committed:
added handler for decimal object in response

Zlash65 committed:
[Minor] Refresh list header on refreshing list_view or performing operations (#5489)

* refresh list header on refresh

* remove redundant flag

nabinhait committed:
Don't allow submitting/cancelling already submitted/cancelled docs (#5484)

rmehta committed:
[minor] ignore_validate option while creating custom fields

surajshetty3416 committed:
[Minor] Fix bug with build_match_condition
- change in commit fb25fdb6 introduced this bug which

Zlash65 committed:
use get_data to override data list

shreyashah115 committed:
map color field

manassolanki committed:
remove bs3 js from the frappe-web.min.js and include that in base.html

codingCoffee committed:
removed bad code and made compatible with pip 10 (#5458)

netchampfaris committed:
Validate fieldnames that contain space (#5499)

Zlash65 committed:
restore original permissions for standard and custom doctypes (#5498)

Zlash65 committed:
hide user-progress and help for mobile view (#5492)

codingCoffee committed:
i still don't know why we're still using travis (#5500)

shreyashah115 committed:
set color if color field in any doctype

nabinhait committed:
Minor fixes

pratu16x7 committed:
[control] code value field

netchampfaris committed:
get_emails_from_role should only pickup enabled users (#5504)

spa49 committed:
Hotfix to allow for modal to be scrolled (#5486)

* Update desk.css

* Update desk.less

saurabh6790 committed:
Revert "Hotfix to allow for modal to be scrolled (#5486)"

This reverts commit 526ebb71a32b3f28daaa50b899f609bfe35dbb22.

saurabh6790 committed:
bumped to version 10.1.26

elba7r committed:
doc-img-fix (#5088)

rohitwaghchaure committed:
Double encode url issue

surajshetty3416 committed:
[urgent] Fix query_report has_match

saurabh6790 committed:
bumped to version 10.1.27

codingCoffee committed:
dark magic to revive travis

It seems that some process inside travis is using port no 9000. Hence
adding this line in the travis.yml to change common_site_config.json and
use port no 9001 for socket_io

netchampfaris committed:
Handle row which can be dict or list

- (why no normalization?)

saurabh6790 committed:
bumped to version 10.1.28

nabinhait committed:
minor fix

netchampfaris committed:
Show footer in Full Page printview

Zlash65 committed:
minor requested changes

surajshetty3416 committed:
Permission improvements (#5494)

* Overwrite "if owner" restriction*
- Overwrite "if owner" restriction if user has other valid roles
(with access rights) without if_owner enabled for it in doc perm
eg:
If a doctype has 2 docperm for Role A and Role B and the docperm with
Role A has  if owner enabled.

Suppose User 1 has Role A, then User 1 should see only docs
which was created by him.

Now if User 2 has Role A and Role B, User 2 should be able to see all the
records because Role B has unrestricted access on doctype.
------------------------------------------------------

Previously in case of User 2, the restriction(of if owner) from Role A
used to reflect and User 2 was only able to view only records made by
him.

This commit will overwrite the if_owner restriction in such case.
- Refactor get_role_permission method

* Add advanced control for user permission
- Add ability to skip user permission check for specific doctype

* Refactor "linked with" method
- Add option to skip linked with doctype which has
ignore_user_permission enabled against them.

* Clear "linked with doctype" cache on doctype changes

* [Minor] Fix formatting

* Fix error and bugs in linked_with method

* Format code

* Add fix to get allowed docs from user_permission

* Fix has_user_permission bug

* Add before migrate patch for user permission

* Fix typo

Commit:
time taken by the patch along with success message (#5524)

ci2014 committed:
Check for list_renderer before accessing settings (#5469)

* Check for list_renderer before accessing settings

In the master version of frappe, in the first attempt of calling the file manager (when onload is called) the report doesn't build, because in the report view, there is no such attribute list_renderer available.

In that change we check if the list renderer is not undefined and only setup the menu, drag and drop and copy function, when the list_renderer object is available.

This problem might not occur in the developer versions (datatables), but many people still work with the master version and need a working file manager report.

* Update file_list.js

codingCoffee committed:
Login fix for python3 (#5529)

decode bytes to string

codingCoffee committed:
pip fix (#5527)

surajshetty3416 committed:
Permission fixes (#5530)

* Add User Permissions to Config module

* Update Permission Manager help

* Fix share button

* Remove user-permissions page

* Remove unnecessary error modals

* User Permission UI fixes
- Fix double multicheck bug on save
- Add margin above and below select all button

* Update user_permissions dict realtime on client side

- Remove user_permissions from boot

* Cast permission dict values to int

nabinhait committed:
minor fix

nabinhait committed:
bumped to version 10.1.29

codingCoffee committed:
Rebuild help for docs which have been renamed (#5521)

After execution of a bulk rename, the help cache for the particular
doctype didn't used to change. Added this fix to ensure the help is
rebuilt for the particular doctype whose docs have been renamed

saurabh6790 committed:
escape new line char while creating boilerplate for app

surajshetty3416 committed:
Fix auto dropdown for tag on page load (#5532)

* Fix auto dropdown for tag on page load

* Fix tag related bugs

- reset doc user_tags on change
- move "label conversion toTitle" from onTagAdd listener
to addTag function
- remove unnecessary refresh call

mnatalia committed:
Update new_message.html (#5535)

Added possibility to translate.

shreyashah115 committed:
Rename occurrences for Production Order to Work Order (#5534)

nabinhait committed:
Increase the length of Singles table columns (#5540)

codingCoffee committed:
added floor and ceil to frappe.utils (#5536)

Added frappe wrappers to math.ceil and math.floor, since they are widely
used functions. Also added their docstrings for documentation and wote
test cases. And some minor typo fixes

saurabh6790 committed:
[fix] set no_copy property for workflow state field (#5531)

* [fix] set no_copy property for workflow state field

* [fix] check if custom field exists before loading doc

nabinhait committed:
bumped to version 10.1.30

codingCoffee committed:
Fix python3 issue reading pdf (#5541)

File types other than plain text files weren't sent properly in email. The initial implementation of try except used to destroy the file object in the try statement, and hence we were getting a blank string in the except block. The fix involves reading the file object separately before trying to decode it.

sushantnadkar committed:
Show fieldname as tooltip for all users if developer_mode=1 (#5543)

codingCoffee committed:
optimized addr_and_contact report by @nabinhait

Old implemetation involved 2 sql queries for every party_type, one for
address and one for contact. This was timing out if the party_type size
was too large. New implementation involves 3 queries only.

Stats:
	no_of_records | party_type | old_time | new_time
	--------------+------------+----------+---------
	27000         | Customer   | 209 secs | 1.5 sec

rmehta committed:
[minor] dont render .min.js files as jinja templates

pratu16x7 committed:
[purge] remove jsbarcode lib, import

netchampfaris committed:
Quick Kanban Board

- dont create project implicitly, results in Deadlock most of the time
- add explicit field to set the project

Zlash65 committed:
[Fix] Email and Contact fixes (#5519)

* use multiselect for email dialog, improv email fetch query

* patch to create contact for all user

* append number if same name found for company

* update tests

netchampfaris committed:
fix codacy

saurabh6790 committed:
bumped to version 10.1.31

surajshetty3416 committed:
Replace frappe.db.sql to frappe.get_list to apply permissions (#5556)

* Replace frappe.db.sql with frappe.get_list
- Permission check for treeviews

* Optimize get_role_permissions
- Avoid setting of roles each time when is_perm_appicable is called

* Fix codacy
- Remove trailing whitespace

codingCoffee committed:
Python 3 fixes for PR bot (#5554)

In python 2 writing to file can be done by reading the file as 'w' or
'wb', however, in python 3 if we're writing 'utf-8' encoded stuff to a
file, it needs to be opened using the 'wb' argument. 'b' here stands for
binary.

surajshetty3416 committed:
Fix a bug with whitelisted method (#5551)

- Add additional check to avoid wrong user set

frappe-pr-bot committed:
[Translation] Updated Translations (#5552)

aliveghost04 committed:
Fix "frappe.is_user_logged_in" js function (#5548)

* Fix is_user_logged_in js function

* Add missing semicolon

yogeshak committed:
JWT response does not contain email attribute for Office 365, reading upn instead (#5525)

rmehta committed:
[minor] remove in_dialog old property

rmehta committed:
Revert "Replace frappe.db.sql to frappe.get_list to apply permissions" (#5559)

* Revert "[minor] remove in_dialog old property"

This reverts commit b4650f42a1b7d29a88bfeb97b464e95b3c79913f.

* Revert "JWT response does not contain email attribute for Office 365, reading upn instead (#5525)"

This reverts commit 51f5988a76e1d112dcbaa59662a8e10008b0bd26.

* Revert "Fix "frappe.is_user_logged_in" js function (#5548)"

This reverts commit affe5b8659edc97a6ea6ff01fece45dfc492e556.

* Revert "[Translation] Updated Translations (#5552)"

This reverts commit f3930a70267bd389b726a9eba85e3781246846dc.

* Revert "Fix a bug with whitelisted method (#5551)"

This reverts commit 92e4db1dd95b2c1aaff53dffd6298a3411c40e63.

* Revert "Python 3 fixes for PR bot (#5554)"

This reverts commit 7b003a71bc0340652a52a2493f35c4f6334eaf36.

* Revert "Replace frappe.db.sql to frappe.get_list to apply permissions (#5556)"

This reverts commit 2e52177f51dc6abedc59419f1b9517b0079c40a5.

surajshetty3416 committed:
Allow everything for Admin (#5561)

- Return every available roles if the user is admin
This will allow admin to access doctype even if doctype
has docperm with only disabled roles
- No more need to add roles for admin on every new role creation

Zlash65 committed:
refresh meta filter for DocType and DocField after column drop

netchampfaris committed:
Datatable fixes (#5569)

- duplicate Add column dropdown item
- Add column dialog should show Label instead of fieldname

chdecultot committed:
Calendar connector improvements (#5571)

* Calendar connector improvements

* Development cleanup

rmehta committed:
[minor] removed duplicate notify_by_email

vjFaLk committed:
Function to remove a slide from the list of slides (#5570)

Zlash65 committed:
[Fix] Sort data when added from quick entry in the list (#5575)

* sort data in list
use sort_by and sort_order individually instead of combining them into one order_by

* patch to update user_settings
- remove order_by and insert sort_by and sort_order in user_settings

netchampfaris committed:
Listview fixes (#5577)

- update sort_by and sort_order on sort_selector change

netchampfaris committed:
[fix][patch] Create contact for user (#5578)

rohitwaghchaure committed:
[Fix] GSTIN number not showing in the company address

Zlash65 committed:
update patch - fix order_by for older settings (#5580)

netchampfaris committed:
Listview fixes (#5581)

* Remove annoying message when kanban is changed

* [fix] view_user_settings getter

* Set last_view on kanban view change

pratu16x7 committed:
[charts] update, without new build system

sharathc3 committed:
Add Custom Jinja methods from hooks.py (#5585)

netchampfaris committed:
Reportview fixes (#5582)

* Reportview- Dynamic row height disabled by default, enable using checkbox

* Always show filter row in Query Report

pratu16x7 committed:
[docs-portal] Search the Docs (#5558)

* [docs-portal] search documentation

* [doc-search] remove search-result template

* [portal-search][minor] add search result section title

* [portal-search] make search result section macro

* [portal-search] two-col layout for more sections

tarokobear committed:
Account for documents with integer names so they can be deleted. (#5567)

saurabh6790 committed:
[fix] handle worker timeout issue while taking backup to dropbox (#5450)

ci2014 committed:
Use email instead of user name for building exclude list from thread_notify (#5502)

* Update email.py

* Get the correct Email Address for Administrator

Get the correct Email Address for the Administrator User as well. I see no reason why.

kickapoo committed:
Select fields to allow in quick entry (#5461)

codingCoffee committed:
bumped to version 10.1.32

pratu16x7 committed:
[query-perf-log] Add execution time for debug opt in frappe.db.sql (#5592)

* [query-perf-log] Add execution time for debug opt in frappe.db.sql

* Update database.py

Zlash65 committed:
[Enhance] New DocField - 'Fetch From' (#5563)

* add new docfield fetch_from - serves as add_fetch

* replace options.split with fetch_from.split

* copied data from options to fetch_from

* patch to update fetch_from for docfield, custom_field and property_setter

* update tests according to fetch_from change

sushantnadkar committed:
[fix] Button duplication on page trigger: refresh (#5564)

* [fix] custom button duplication on page trigger: refresh

* added data-label to buttons to filter them more easily

* added encodeURIComponent to data-label

* Update page.js

chdecultot committed:
Allow several Stripe accounts (#5573)

* Allow several Stripe accounts

* Addition of a sync patch

* Remove unused dependancy

gaurav-naik committed:
modified address and contact js (#4862) (#5586)

* modified aaddress and contact js

- temporary fix to handle route_history if docname is naming_series with a '/' in the name- permanant fix would be to modify route options
- PR linked to frappe/erpnext#12215

* considered cases with more '/'s in doc name

rohitwaghchaure committed:
[Fix] Hyperlink showing in the pdf print (#5590)

Zlash65 committed:
run docfield patch prior to other patches (#5594)

nabinhait committed:
Update activity_log.json

rmehta committed:
[minor] add pdfkit to requirements.txt

rmehta committed:
[fix] ellipsis in navbar brand #5599

rmehta committed:
[fix] calendar render for empty list, user_list.js

codingCoffee committed:
[fix] global-setup-help (#5603)

- custom jinja templates are taken if and only if a site is mentioned
- added test in travis as well

codingCoffee committed:
Prevents circular linking of docs (#5589)

* Prevents circular linking of docs

- also added test case to check if code to prevent circular linking of
doctypes works

* created a CircularLinkingError class

- this can be used later if circular linking error needs to be thrown
elsewhere

Zlash65 committed:
fix email query (#5605)

codingCoffee committed:
bumped to version 10.1.33

sharathc3 committed:
Jinja Customization from hooks.py bug fix (#5604)

* Jinja Customization from hooks.py bug fix

* Condition to check for site initialization added.

nabinhait committed:
fetch_from fixes

nabinhait committed:
fetch_from fixes in mapper

nabinhait committed:
Fetch From also applicable for Check fields (#5611)

Commit:
handle Int overflow (#5607)

* handle the int overflow

 - convert the int to the bigint if the length is more than 11

* proper formatting

schilgod committed:
Enhancement to allow User to login from any IP if two factor auth is enabled (#5209)

* Enhancement to allow login from any IP if two factor auth is enabled

* Resolve Conflicts

* optimize code

pratu16x7 committed:
[model][std-fields] make 'Created By' a User Link field (#5600)

rmehta committed:
[minor] strip special characters when making a fieldname

pratu16x7 committed:
[fix] cached timestamp in frappe.model.with_doctype() (#5613)

* [fix] cached timestamp in frappe.model.with_doctype()

* Update model.js

shridarpatil committed:
add new doctype to store view logs

yimesgen committed:
Update country_info.json (#5616)

surajshetty3416 committed:
Treeview permissions (#5615)

* Replace frappe.db.sql to frappe.get_list to apply permissions (#5556)

* Replace frappe.db.sql with frappe.get_list
- Permission check for treeviews

* Optimize get_role_permissions
- Avoid setting of roles each time when is_perm_appicable is called

* Fix codacy
- Remove trailing whitespace

* Add parent filter

* Add ifnull check

codingCoffee committed:
[fix] contact to user creation patch (#5617)

- special charcters in name used to break user creation, this patch
prevents user creation in case of bad characters

rmehta committed:
[fix] for bench get-app

codingCoffee committed:
[fix] creates contact even in presence of bad char (#5618)

saurabh6790 committed:
bumped to version 10.1.34

rmehta committed:
[fix] bench_helper.py

netchampfaris committed:
fix: Don't add currency field if it does not exist (#5619)

closes #4995

Commit:
Update db_schema.py (#5622)

netchampfaris committed:
fix: get_contact_list in Email dialog (#5623)

shridarpatil committed:
doctype for view logs

shridarpatil committed:
client side changes

nabinhait committed:
Fetch from property is now applicable for all fieldtypes except no_value_fields (#5624)

* Reload docperm

* Reload docperm

* Fetch from property is now applicable for all fieldtypes except no_value_fields

nabinhait committed:
changed patch order

shridarpatil committed:
view log doctype updated

shridarpatil committed:
add to view_log whenever a user views a document

shridarpatil committed:
view log changed fields to set only once

shridarpatil committed:
get and retun view logs if available

shridarpatil committed:
add view logs to time line
